ddtrace/mocktracer/mockspan.go

```diff
@@ -66,7 +66,10 @@ func newSpan(t *mocktracer, operationName string, cfg *ddtrace.StartSpanConfig)
 	} else {
 		s.startTime = cfg.StartTime
 	}
-	id := nextID()
+	id := cfg.SpanID
+	if id == 0 {
+		id = nextID()
+	}
```

Comment:
This is a bit tricky. Imagine the following scenario:
* You create a span with a custom ID of 2
* You create a regular span with a generated ID (by `nextID`), it gets ID 1
* You create another one, it gets ID 2 and clashes with our customly defined ID.

To have this working correctly, we'll have to create a mock generator that uses locks, and update the source used to the user specified ID when it's larger than it, if that makes sense.

It seems minor and something we can live with, but I can imagine that if it becomes a problem, it will be hard to figure out.

Reply:
Or, if you can think of an easier way (there might be one), I'm open to suggestions.

Reply:
Good catch, did not think of that.
So any change to spanID generation is an API breaking change, are you guys fine with that?
If so, why not just randomize the spanID generation?

So you would have a something like this?
```
type idSource {
        id uint64
        mutex sync.Mutex
}

func (s *idSource) next(given uint64) uint64 {
       mutex.Lock()
       defer mutex.Unlock()
        if given == 0 {
                s.id += 1
        } else {
                s.id = given        
        }
        return s.id
}

var mockIdSource = &idSource{123}
```
(could be split up into `seed(uint64)` and `next(uint64)`)

I like your suggestion but I have one more (which might be less of a breaking change):

Have the mocktracer store a set of the id's used, and let `nextID()` increment until a non-used id is found. If memory is of concern, a bloom filter could do aswell.

Reply:
Thinking about it more, I think it complicates things too much to do this. The map version you've suggested is likely the easiest approach but might be overkill for the sake of testing. I'd say it's fine to ignore this for now.